storage/tianmu/core/query_compile.cpp

```diff
@@ -919,51 +920,48 @@ int Query::Compile(CompiledQuery *compiled_query, SELECT_LEX *selects_list, SELE
   CompiledQuery *saved_cq = cq;
   cq = compiled_query;
 
-  if ((selects_list->join)&&(selects_list != selects_list->join->unit->global_parameters())) {  // only in case of unions this is set
+  if ((selects_list->join) &&
+      (selects_list != selects_list->join->unit->global_parameters())) {  // only in case of unions this is set
     SetLimit(selects_list->join->unit->global_parameters(), 0, global_offset_value, (int64_t &)global_limit_value);
     global_order = &(selects_list->join->unit->global_parameters()->order_list);
   }
 
   for (SELECT_LEX *sl = selects_list; sl; sl = sl->next_select()) {
     int64_t limit_value = -1;
     int64_t offset_value = -1;
+    bool isNewJon = false;
+    if (!sl->join) {
+      sl->add_active_options(SELECT_NO_UNLOCK);
+      JOIN *join = new JOIN(sl->master_unit()->thd, sl);
+
+      if (!join) {
+        sl->cleanup(0);
+        return TRUE;
+      }
+      isNewJon = true;
```

Comment:
What is var `isNewJon` mean?  ---> `isNewJoin`?
Add comment to desc this var.
or use `ifNewJoinForTianmu` ?